### PR TITLE
Fix workflow

### PR DESCRIPTION
**Pull request by Aavild**
- Fixed old deprecated action to use newer versions
 - Did some minor renaming and refactoring on the actions
 - ci-yaml -> check_format.yaml & run_tests.yaml
 - fixed some tests as dictated by Clippy

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (williamwoldum) [inline .github/workflows/ci.yaml:1]: `types: [opened, reopened, synchronize]` is the default behavior of `pull_request`and can be omitted.
 https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Review by williamwoldum — APPROVED** [review]
👌

File: .github/workflows/ci.yaml
Comment (Aavild) [reply .github/workflows/ci.yaml:1]: For readabilitys sake this will not be omitted

**Review by williamwoldum — APPROVED** [review]

**Review by williamwoldum — COMMENTED** [review]
These changes will have `build_artifacts` running as a check on every PR to `main`.
Is this intended behavior?

I assume we only want to run `build_artifacts` after all checks have passed, and all changes have been merged into `main`.

**Review by MadsSR — APPROVED** [review]

**Review by rasmusq — APPROVED** [review]

**Review by williamwoldum — APPROVED** [review]

**Pull request merged**